std::os::fd module missing on Hermit

I tried this code:

```rust
use std::os::fd::{AsFd, AsRawFd, BorrowedFd, FromRawFd, IntoRawFd, RawFd};
``` 

With `--target x86_64-unknown-hermit`.

I expected to see this happen: code to compile

Instead, this happened: failed to compile with

```
error[E0432]: unresolved import `std::os::fd`
   --> src/net/udp.rs:12:14
    |
12  | use std::os::fd::{AsFd, AsRawFd, BorrowedFd, FromRawFd, IntoRawFd, RawFd};
    |              ^^ could not find `fd` in `os`
```

Looking at the relevant code: https://github.com/rust-lang/rust/blob/212841e17cce13d1e29b00ad144300b687bfafdd/library/std/src/os/mod.rs#L163-L164

It seems `hermit` is imply missed from the `cfg`. Inside of the `fd` module there is already code for hermit: https://github.com/rust-lang/rust/blob/212841e17cce13d1e29b00ad144300b687bfafdd/library/std/src/os/fd/raw.rs#L18-L29

I think this is just an oversight, but I since I don't develop for Hermit I'm not sure. (I hit this while updating Mio)

/cc @stlankes

### Meta

`rustc --version --verbose`:
```
rustc 1.81.0-nightly (f21554f7f 2024-06-08)
binary: rustc
commit-hash: f21554f7f0ff447b803961c51acafde04553c1ed
commit-date: 2024-06-08
host: x86_64-unknown-linux-gnu
release: 1.81.0-nightly
LLVM version: 18.1.7
```

--- a/library/std/src/os/hermit/io/mod.rs
+++ b/library/std/src/os/hermit/io/mod.rs
@@ -1,13 +1,4 @@
-#![stable(feature = "os_fd", since = "1.66.0")]
+#![stable(feature = "rust1", since = "1.0.0")]
 
-mod net;
-#[path = "../../fd/owned.rs"]
-mod owned;
-#[path = "../../fd/raw.rs"]
-mod raw;
-
-// Export the types and traits for the public API.
-#[stable(feature = "os_fd", since = "1.66.0")]
-pub use owned::*;
-#[stable(feature = "os_fd", since = "1.66.0")]
-pub use raw::*;
+#[stable(feature = "rust1", since = "1.0.0")]
+pub use crate::os::fd::*;

--- a/library/std/src/os/mod.rs
+++ b/library/std/src/os/mod.rs
@@ -160,7 +160,7 @@ pub(crate) mod watchos;
 #[cfg(target_os = "xous")]
 pub mod xous;
 
-#[cfg(any(unix, target_os = "wasi", doc))]
+#[cfg(any(unix, target_os = "hermit", target_os = "wasi", doc))]
 pub mod fd;
 
 #[cfg(any(target_os = "linux", target_os = "android", doc))]

